trikotFoundation/src/iosArm32Main/kotlin/com/mirego/trikot/foundation/helpers/ByteArrayNativeUtils.kt

```diff
@@ -1,22 +1,20 @@
 package com.mirego.trikot.foundation.helpers
 
+import kotlinx.cinterop.ByteVar
+import kotlinx.cinterop.CPointer
 import kotlinx.cinterop.addressOf
+import kotlinx.cinterop.get
+import kotlinx.cinterop.reinterpret
 import kotlinx.cinterop.usePinned
 import platform.Foundation.NSData
 import platform.Foundation.create
-import platform.posix.memcpy
 
 object ByteArrayNativeUtils {
     @ExperimentalUnsignedTypes
     fun convert(data: NSData): ByteArray {
-        val byteArray = ByteArray(data.length.toInt()).apply {
-            usePinned {
-                memcpy(it.addressOf(0), data.bytes, data.length)
-            }
-        }
-        return byteArray
+        val dataPointer: CPointer<ByteVar> = data.bytes!!.reinterpret()
```

Comment:
`!!` 👀 It could be nice to not force an unwrapping

Reply:
Ouin j'avoue, un peu trop de trust envers la gang de ktor

De même mettons:

```
fun convert(data: NSData): ByteArray {
        return data.bytes?.let {
            val dataPointer: CPointer<ByteVar> = it.reinterpret()
            ByteArray(data.length.toInt()) { index -> dataPointer[index] }
        } ?: ByteArray(0)
    }
```